mix.exs

```diff
@@ -152,6 +152,40 @@ defmodule Mix.Tasks.Compile.Tika do
     end
   end
 
+  defp proxy_auth(url) do
+    # url scheme
+    url
+    |> get_scheme
+    |> get_proxy_url
+    |> get_proxy_auth_from_proxy_url()
+  end
+
+  defp get_scheme(url) do
+    cond do
+      String.starts_with?(url, "http://") -> :http
+      String.starts_with?(url, "https://") -> :https
+    end
+  end
+
+  defp get_proxy_url(:http) do
+    System.get_env("HTTP_PROXY") || System.get_env("http_proxy")
+  end
+
+  defp get_proxy_url(:https) do
+    System.get_env("HTTPS_PROXY") || System.get_env("https_proxy")
+  end
+
+  defp get_proxy_auth_from_proxy_url(proxy_url) do
+    parsed = URI.parse(proxy_url)
```

Comment:
The build is failing because the proxy URl can be `nil` when there are no environment variables set; do you mind adding either an `if` statement or a guard to this function to handle that case?

Reply:
oh yes, ofcourse. Will do this asap.

Reply:
added the check to handle the case where neither `http_proxy` nor `https_proxy` env variable are present